### ISSUE TITLE
Questions about custom dataset

### ISSUE THREAD

**Issue by quanw8781** [trqminh/aistron#3]
Thank you for your excellent work and I have some questions.
Is it possible to train and predict using my own data, and if so, what should I do about it? I'm using the Labelme tool to label the images. How can I convert the json files to COCOA format or any other format that works? 
Looking forward to your reply.

**Comment by trqminh** [comment]
Yes, it's possible to train with your custom data. However, we're currently not supporting training with custom dataset directly. You can still work around by doing the following steps:
- convert your labelme annotation to one of the supported dataset format (e.g. KINS, D2SA, or COCOA)
- use the corresponding prepare scripts [here](https://github.com/trqminh/aistron/tree/main/datasets) to convert the annotation to aistron universal annotation
- lastly, you're gonna need to register you dataset, you can refer from [here](https://github.com/trqminh/aistron/tree/main/aistron/data/datasets)

I'll try to add custom dataset support in the near future, but in the mean time, you can work around based on the above steps.

**Comment by binh0804** [comment]
Thank you for dedicating the time to this, hope to see your reply soon. 
Sorry, can you share how labelme (or another label tool) can label occluded part of objects? I am preparing custom dataset based on KINS dataset annotation format and face to some problem. There are some fields that I don't know how to label such as "**a_segm**","**i_segm**" in update_test_2020.json or "**amodal_full**" in instances_train.json. Or, I have to edit the annotations file manually?

**Comment by CHN-001** [comment]
> 感谢您花时间做这件事，希望很快收到您的回复。 抱歉，您能分享一下 labelme（或其他标签工具）如何标记物体的遮挡部分吗？我正在根据 KINS 数据集注释格式准备自定义数据集，但遇到了一些问题。有些字段我不知道如何标记，例如 update_test_2020.json 中的“ **a_segm** ”、“ **i_segm** ”或 instance_train.json 中的“ **amodal_full** ”。或者，我必须手动编辑注释文件？

你好，我遇到了同样的问题，我现在有COCO格式的自建数据集，或者是labelme标注生成的json格式文件，但是这些文件没有关于Amodal的信息，这导致了在复现模型的时候无法评估非模态分割的性能。我应该如何制作COCOA格式呢？或者能否将已有的COCO格式的数据转换成COCOA格式的数据？

**Comment by binh0804** [comment]
(Sorry for reply in Vietnamese because you use Chinese)
Nhóm chúng tôi thực hiện các bước tương tự theo hướng dẫn sau: https://git.wur.nl/blok012/sizecnn/-/blob/master/ANNOTATE.md?ref_type=heads và sau đó chạy file labelme_to_orcnn.py (của repository đó). File annotation dưới dạng json lúc này đã có thể dùng với prepare_cocoa.py của aistron. Chúc may mắn!

**Comment by CHN-001** [comment]
> (Sorry for reply in Vietnamese because you use Chinese) Nhóm chúng tôi thực hiện các bước tương tự theo hướng dẫn sau: https://git.wur.nl/blok012/sizecnn/-/blob/master/ANNOTATE.md?ref_type=heads và sau đó chạy file labelme_to_orcnn.py (của repository đó). File annotation dưới dạng json lúc này đã có thể dùng với prepare_cocoa.py của aistron. Chúc may mắn!
Thank you very much indeed. You have been a great help to me. I will try to follow your method, but before I try, I have a few questions to ask you. First, if half of my tagged object is obscured and half is visible, should I label the visible half as visible and the whole as amodal? Second, if the entire amodal object is labeled with polygon, will human imagination affect the performance and effect?
I will try it first. If I encounter any problems, I hope you can help me again. I really appreciate your help!

**Comment by CHN-001** [comment]
> (Sorry for reply in Vietnamese because you use Chinese) Nhóm chúng tôi thực hiện các bước tương tự theo hướng dẫn sau: https://git.wur.nl/blok012/sizecnn/-/blob/master/ANNOTATE.md?ref_type=heads và sau đó chạy file labelme_to_orcnn.py (của repository đó). File annotation dưới dạng json lúc này đã có thể dùng với prepare_cocoa.py của aistron. Chúc may mắn!

Thank you very much indeed. You have been a great help to me. I will try to follow your method, but before I try, I have a few questions to ask you. First, if half of my tagged object is obscured and half is visible, should I label the visible half as visible and the whole as amodal? Second, if the entire amodal object is labeled with polygon, will human imagination affect the performance and effect?
I will try it first. If I encounter any problems, I hope you can help me again. I really appreciate your help!

**Comment by binh0804** [comment]
First: In my case, crystal, I label the visible part with class "crystal_visible" and set tag 1, and I imagine the complete shape of this crystal => label it "crystal_amodal" and set tag 1 also. Repeat to crystal 2,3,4,5, etc.
Second: yes human imagination affect the performance. Of course, the machine learned what you trained it. 
You can contact me at binh08042002@gmail.com, I am also doing graduate project ơn this topic.

**Comment by CHN-001** [comment]
> First: In my case, crystal, I label the visible part with class "crystal_visible" and set tag 1, and I imagine the complete shape of this crystal => label it "crystal_amodal" and set tag 1 also. Repeat to crystal 2,3,4,5, etc. Second: yes human imagination affect the performance. Of course, the machine learned what you trained it. You can contact me at [binh08042002@gmail.com](mailto:binh08042002@gmail.com), I am also doing graduate project ơn this topic.

Thank you very much for your answer, which is very clear and has solved my problems. I will contact you by email if I have any questions in the future! Thank you for your help!

**Comment by Vincent-Songwentao** [comment]
> 是的，您可以使用自定义数据进行训练。但是，我们目前不支持直接使用自定义数据集进行训练。您仍然可以通过以下步骤来实现：
> 
> * 将您的 labelme 注释转换为受支持的数据集格式之一（例如 KINS、D2SA 或 COCOA）。
> * [使用此处的](https://github.com/trqminh/aistron/tree/main/datasets)相应准备脚本将注释转换为 Aistron 通用注释。
> * 最后，您需要注册您的数据集，您可以参考[这里。](https://github.com/trqminh/aistron/tree/main/aistron/data/datasets)
> 
> 我会在不久的将来尝试添加自定义数据集支持，但在此期间，您可以根据上述步骤进行变通。

请问注册自己数据集的详细流程是什么
